native/src/main/java/io/ballerina/stdlib/mqtt/client/ClientActions.java

```diff
@@ -107,8 +108,9 @@ public static Object externSubscribe(BObject clientObject, BArray subscriptions)
     public static Object externPublish(Environment env, BObject clientObject, BString topic, BMap message) {
         MqttClient publisher = (MqttClient) clientObject.getNativeData(MqttConstants.MQTT_CLIENT);
         MqttMessage mqttMessage = generateMqttMessage(message);
-        try {
-            Future future = env.markAsync();
+        return env.yieldAndRun(() -> {
+            CompletableFuture<Object> future = new CompletableFuture<>();
```

Comment:
i think we can remove publishExecutorService.execute and directly use deliveryTokenQueue.take(). So then we can remove Util class as well.